# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/alberttigo24-droid/Albert-Flyrank-assignments/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

## 1. My lane (or freestyle) and why

**Provisional lane: Lane 2 — Refresh / Content Opportunity Scoring.**

I want to investigate how a content team could prioritize which pages to review first when it has more pages than it can realistically inspect. The output would be a ranked review queue based on observable content, search, and engagement signals. I chose this lane because it connects directly to a real resource-allocation decision, and the starter workflow already provides evidence that a learned ranking may outperform a simple rule on its starter slice. This is a provisional choice: I can change the lane after learning more from the data and early validation.

In [3]:
# Load the starter data and show basic supporting numbers.
# This cell uses only observable starter-data fields.

import pandas as pd

df = pd.read_csv("/content/content_refresh_anonymized.csv")

print(f"Rows: {len(df):,}")
print(f"Unique content pages: {df['content_id'].nunique():,}")
print(f"Unique clients: {df['client_id'].nunique():,}")

declining_pct = (df["trend_direction"].eq("down").mean() * 100)
median_impressions = df["impressions_90d"].median()
visible_500_pct = (df["impressions_90d"] >= 500).mean() * 100

print(f"Pages with downward current-window trend: {declining_pct:.1f}%")
print(f"Median 90-day impressions per page: {median_impressions:,.0f}")
print(f"Pages with at least 500 90-day impressions: {visible_500_pct:.1f}%")

Rows: 30,000
Unique content pages: 30,000
Unique clients: 32
Pages with downward current-window trend: 54.2%
Median 90-day impressions per page: 731
Pages with at least 500 90-day impressions: 55.8%


## 2. The question: decision, action, cost of a wrong call

**Research question:** When review capacity is limited, which content pages should be reviewed first based on observable evidence?

- **Decision:** Decide which pages receive review attention first.
- **Who acts:** A content/SEO reviewer or editor uses the ranked queue to decide where to spend limited review time.
- **Action:** The reviewer may choose to refresh, expand, protect, prune, or monitor a page after inspecting the evidence.
- **Unit of analysis:** One content page (`content_id`).
- **Output:** A ranked priority score/queue, ideally with reason codes that explain why a page is high priority.
- **Cost of a wrong call:** A false positive can waste scarce reviewer time on a page that was not a good candidate. A false negative can cause a useful review opportunity to be missed. The relative cost of these errors should influence the final ranking threshold and validation metric.

This is a ranking/decision-support problem rather than simply a request to predict whether a page is "bad." The goal is to improve the order in which humans review pages, not to automate the final content decision.

## 3. Quick look at the data (2-3 real numbers)

The starter dataset contains **30,000 content pages** across **32 clients**. In this slice, **54.2%** of pages have a downward `trend_direction`, and the median page has **731 90-day impressions**. **55.8%** of pages have at least 500 90-day impressions.

These numbers make the lane worth investigating because there is a substantial inventory of pages to prioritize, and many pages have measurable search exposure. The downward-trend figure is only a current-window signal, not proof that a page will continue to decline or that a refresh would cause recovery. The 500-impression figure also does not mean every such page deserves action; it simply shows that a meaningful subset has enough observed exposure to make prioritization potentially useful.

The code cell above calculates these values directly from the starter CSV.

## 4. Careful words: what I can and can't claim

### What I can claim

I can use the starter data to measure associations between observable page signals and the starter outcome, and to investigate whether a ranking method can put more useful candidates near the top of a review queue. Results should be described as **observed, directional, and decision-support** findings.

I can also compare a transparent baseline with a learned ranking method using a metric such as Precision@K, because the real operational decision is about which limited number of pages appear first.

### What I cannot claim

I cannot claim that a high-priority page is guaranteed to need a refresh, or that refreshing it will cause traffic, clicks, or engagement to recover. The starter `is_declining_label` is derived from the current `trend_direction`, so it is a proxy rather than a future observed outcome. A stronger later version of this project should define a future-window outcome and keep the feature window separate from that target window.

I also cannot claim that the analysis proves a Google ranking factor, predicts Google's algorithm, or establishes causation. The data is observational. Any later model must be checked for leakage and validated against the actual decision it is intended to support.

## Self-check

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors
- [x] No client names, URLs, or private queries from the dataset are used
- [x] Claims use careful words: observed, measured, directional, decision-support
- [x] The lane is explicitly provisional and can be changed after early evidence
- [x] The output is a ranked review queue, not an automatic "fix this page" decision
- [x] The notebook is ready to commit under `work/notebooks/`